src/commons/UI/star-rating/useBaseStarRating.ts

```diff
@@ -0,0 +1,51 @@
+import { UPDATE_MODEL_VALUE } from "@/commons/constant";
+import { ref, SetupContext } from "vue";
+
+export default function useBaseStarRating(
+  props: Readonly<

```

Comment:
this type looks ugly. Can we just use modalValue as an argument instead?
example here: https://github.com/hu-tao-supremacy/frontend/commit/e88983876e7a9fbcab1d275f17898781dba6a6e9